### DISCUSSION TITLE
Lazy loading in a large bundle with huge number of screens (150-200)

### DISCUSSION THREAD

**Discussion by jayshah-swiggy** [react-navigation/react-navigation#10170] [Q&A] [answered]
Hello I have app with large number of screens, many of which are not needed immeidiately.
We keep the usage simple as mentioned https://reactnavigation.org/docs/stack-navigator/#api-definition
Is there a general recommendation as to improve startup performance with react-navigations.

**Comment by satya164** [comment, marked as answer]
You can use `getScreen` with RAM bundles, but doesn't Hermes already improve startup performance?

https://reactnavigation.org/docs/screen/#getcomponent

**Reply by jayshah-swiggy** [reply]
getComponent solves the problems with hermes, thanks. On top of that I had to use named imports/exports for inline requires to work correctly.

**Reply by pke** [reply]
@jayshah-swiggy could you maybe provide a small snippet how this looks like?

@satya164 when I am using getComponent like this:

`getComponent={() => () => <MarkdownScreen source={require("../terms.md")}/>}` 

then the screen is mounted 3 times (as checked via useEffect) and its also "mounted" when navigating back from the screen.

However when I define the Screen as a function:

```tsx
function TermScreen() {
  return <MarkdownScreen source={require("../terms.md")}/>
}

getComponent={() => TermScreen}
```

Its correctly mounted once and not mounted again when navigated away from it.
Why is that?
